managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java

```diff
@@ -139,8 +139,14 @@ public void initiate() {
             lastInitTime = System.nanoTime();
             if (ml.getManagedLedgerInterceptor() != null) {
                 long originalDataLen = data.readableBytes();
-                payloadProcessorHandle = ml.getManagedLedgerInterceptor()
-                        .processPayloadBeforeLedgerWrite(this.getCtx(), duplicateBuffer);
+                try {
+                    payloadProcessorHandle = ml.getManagedLedgerInterceptor()
+                            .processPayloadBeforeLedgerWrite(this.getCtx(), duplicateBuffer);
+                } catch (Exception e) {
+                    log.error("[{}] Error processing payload before ledger write", ml.getName(), e);
```

Comment:
would it make sense to call `duplicateBuffer.release()` here?

Reply:
+1.
Referring to: https://github.com/apache/pulsar/blob/master/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java#L811-L879

it will call OpAddEntry.failed(..) before OpAddEntry.initiate(). Since OpAddEntry.initiate() retained the buffer, so call `duplicateBuffer.release()` here is reasonable.

Reply:
Good idea.

It makes sense. Normally, the release operation will execute in the Processor's release method, but if exceptions are thrown, the release operation can't be executed, if we don't release data here, the processor needs to release data when throwing exceptions.